python_client/web_api_client.py

```diff
@@ -0,0 +1,135 @@
+    """
+
+    def __init__(self, base_url):
+        self.base_url = base_url
+
+    def upload(self, file_path):
+        """
+        Uploads a file to the web application.
+
+        Args:
+            file_path (str): The local file path to be uploaded.
+
+        Returns:
+            str: The unique identifier (UID) associated with the uploaded file.
+
+        Raises:
+            Exception: If an error occurs during the file upload.
+        """
+        url = f"{self.base_url}/upload"
+        files = {'file': open(file_path, 'rb')}
+        response = requests.post(url, files=files)
+
+        if response.status_code == 200:
+            data = response.json()
+            return data['uid']
+        else:
+            raise Exception(f"Error uploading file: {response.text}")
+
+    def status(self, uid):
+        """
+        Gets the processing status of a file.
+
+        Args:
+            uid (str): The unique identifier (UID) of the file whose status is to be checked.
+
+        Returns:
+            Status: An instance of the Status class representing the file's processing status.
+
+        Raises:
+            Exception: If an error occurs while getting the status.
+        """
+        url = f"{self.base_url}/status/{uid}"
+        response = requests.get(url)
+        if response.status_code == 200:
+            data = response.json()
+            status = data['status']
+            filename = data['filename']
+            timestamp = datetime.strptime(data['timestamp'][:-5], "%Y%m%d%H%M%S")
+            explanation = data.get('explanation', '')
+            return Status(status, filename, timestamp, explanation)
+        else:
+            raise Exception(f"Error getting status: {response.text}")
+
+
+def main(file_path: str = DEFAULT_FILE_PATH,
+         sleep_time: float = 20):
+    """
+    Main function to interact with the web application and check the processing status of a file.
+
+    Args:
+        file_path (str, optional): The local file path to be uploaded. Defaults to "C:\\Users\\User\\Desktop\\excellenteam\\excellenteam_python\\t.pptx".
+        sleep_time (float, optional): The time to sleep till checking for status (in seconds). Defaults to 20 seconds.
+    """
+    base_url = 'http://localhost:5000'
+    client = WebAppClient(base_url)
+
+    # Upload the file
+    uid = client.upload(file_path)
+    print(f"File uploaded. UID: {uid}")
+
+    # Check status
+    try:
+        time.sleep(sleep_time)
+        status = client.status(uid)
+        if status.is_done():
+            print("File processing is done.")
+            print(f"Filename: {status.filename}")
+            print(f"Timestamp: {status.timestamp}")
+            print(f"Explanation: {status.explanation}")
+        else:
+            print("File is still processing.")
+            print(f"Filename: {status.filename}")
+            print(f"Timestamp: {status.timestamp}")
+    except Exception as e:
```

Comment:
Always catch the most specific exception -
https://stackoverflow.com/questions/730764/how-to-properly-ignore-exceptions